labs/a4.md

```diff
@@ -0,0 +1,70 @@
+---
+title: Lab 4 - Post Install
+layout: lab
+---
+
+# Generating SSH Keys
+Since passwords can be guessed, a secure SSH setup will usually use key-based authentication instead of (or in addition to) a passphrase. You may already have an ssh key, so run the following command to see if you have one:
+
+(note: We won’t give instructions here for Windows machines. To follow these instructions, SSH into `ssh.ocf.berkeley.edu` with your OCF account’s credentials and continue from there)
+
+```
+ls .ssh
+```
+
+If you see id_rsa or id_edsca listed, you already have an SSH key. If you don’t, run this command to generate one:
+
+
+```
+ssh-keygen -t rsa -b 4096
+```
+
+Now, run `ls .ssh` again. You should see two files, `id_rsa` and `id_rsa.pub`.
+
+*  `id_rsa` is the private key-- it should never be revealed outside of the machine it’s generated on. Knowledge of this key can be used to enter your server!
+* `id_rsa.pub` is the public key. You can safely give it to your friends and/or shout it from the rooftops. If you put it in the `~/.ssh/authorized_keys` folder of a user on a server, you will be able to log in as that user on that server.
```

Comment:
You will be allowed a *passwordless* login as the user.

Reply:
This isn't necessarily true, since they might set a passphrase on their SSH key.